src/app/notifications/components/Notifications.tsx

```diff
@@ -38,11 +47,13 @@ const renderNotification = (
 
 export class Notifications extends Component<Props> {
   public render() {
-    const { notification, className } = this.props;
-    return notification ? renderNotification(notification, className) : null;
+    const { notifications: {firstInQueue, queuelessNotifications}, className } = this.props;
+    const notificationsToRender = [...queuelessNotifications, firstInQueue || null ];
```

Comment:
you've changed a lot of logic to make this `queuelessNotifications` thing happen this way. in general, i think the logic you wrote looks fine, but this is our first notification that works outside the normal queue, and we have no idea what the requirements will be for future notifications that may or may not work within the queue or may or may not have a _separate_ queue that works with the search error. i just don't think we have enough information yet to make the kinds of design changes that you've made here. 

i think a simpler solution that we can use until we have more use cases is to not handle SearchFailure here in the notifications component. i would _try_ to set something up so that `searchHighlightManager` controls when to display it and hook that up within PageComponent in order to avoid having a notification action for searchFailure at all. if that doesn't work for whatever reason we could have `searchHighlightManager` dispatch something (maybe notification action, but in that case make sure that it is filtered out of `notificationForDisplay`) and have `SearchFailure` internally decide whether to return null or not (keeping it out of any other notification queuing logic) and just unconditionally throw a `<SearchFailure />` wherever it has to be to make the design work